### PR TITLE
config: create all directories

### PR DESCRIPTION
**Pull request by guggero**
Fixes https://github.com/lightningnetwork/lnd/issues/4572.

With this PR we make sure that all directories where lnd could
potentially want to write files to are created on startup. This fixes
the case where the lnddir isn't set but all other paths point to
explicit locations with non-existend parent directories.

We don't create the log dir as that's done by the log rotator already.

### CODE REVIEW

File: config.go [hunk 1]
```diff
@@ -588,6 +593,43 @@ func ValidateConfig(cfg Config, usageMessage string) (*Config, error) {
 	cfg.Tor.WatchtowerKeyPath = CleanAndExpandPath(cfg.Tor.WatchtowerKeyPath)
 	cfg.Watchtower.TowerDir = CleanAndExpandPath(cfg.Watchtower.TowerDir)
 
+	// Create the lnd directory and all other sub directories if they don't
+	// already exist. This makes sure that directory trees are also created
+	// for files that point to outside of the lnddir.
+	if err := makeDirectory(lndDir); err != nil {
```
Comment (cfromknecht) [inline config.go:599]: nit: can use `for` loop :)

File: config.go [hunk 2]
```diff
@@ -550,23 +550,28 @@ func ValidateConfig(cfg Config, usageMessage string) (*Config, error) {
...
+	makeDirectory := func(dir string) error {
+		err := os.MkdirAll(dir, 0700)
+		if err != nil {
+			// Show a nicer error message if it's because a symlink
+			// is linked to a directory that does not exist
+			// (probably because it's not mounted).
+			if e, ok := err.(*os.PathError); ok && os.IsExist(err) {
+				link, lerr := os.Readlink(e.Path)
+				if lerr == nil {
+					str := "is symlink %s -> %s mounted?"
```
Comment (cfromknecht) [inline config.go:563]: 👍

**Review by cfromknecht — COMMENTED** [review]
nice fix!

**Review by halseth — APPROVED** [review]
Nice, LGTM 👍

File: config.go (hunk 1 above)
Comment (guggero) [reply config.go]: Good idea, done.

**Review by cfromknecht — APPROVED** [review]
LGTM 💪

**Pull request merged**